### DISCUSSION TITLE
Add multiple Query Parameters with the same name to a Request

### DISCUSSION THREAD

**Discussion by stankobrin** [saloonphp/saloon#215] [Q&A] [answered]
Hi All

I am integrating to an API that requires you to add an array of all custom_fields using a single key name. i.ie:
`properties=field_name1&properties=field_name2`

Whenever I try and use the `$request->query()` method, the API only returns the **last** item and not both.
I have tried the `->merge()` but there is no documentation on this and again it just returns either none of the items or only the last.

Examples I have tried:
```
$request->query()->add('properties', 'field_name1');
$request->query()->add('properties', 'field_name2');
```

```
$properties = ['field_name1', 'field_name2'];
$request->query()->add('properties', $properties);
```

```
$properties_array = [ 'properties' => 'field_name1',  'properties' => 'field_name2',];
$request->query()->merge($properties_array);
```

```
$properties_array = [ 'properties' => 'field_name1',  'properties' => 'field_name2',];
$request->query()->add('properties', $properties_array);
```

Any idea what to do here?

Thanks
Stan

**Comment by Sammyjo20** [comment, marked as answer]
Hey @stankobrin many thanks for your comment. 

I'll be honest and I never considered multiple value query parameters in Saloon, but now you mention it I wish I had it in core. What if I created a PR which allows you to provide an array as the `value` and then each item in the array is imploded with a `,` ? So you end up with something like:

`properties=field_name1,field_name2`

You can get around this by writing your own `implode()` method:

```php
$request->query()->add('properties', implode(',', ['value1', 'value2']));
```

**Reply by stankobrin** [reply]
That would be perfect! and the work around works too!
